policies/api-compat.md

```diff
@@ -0,0 +1,19 @@
+Policy on API compatibility in minor releases
+=============================================
+
+**No changes to existing public API functions and data are permitted.** This
```

Comment:
Do we need a definition of what constitutes a "public API function" or "data"?

Reply:
I don't know. It should obviously be anything declared in the header files in `include/openssl`. Anything else?

Reply:
Probably that is sufficient. I do think it needs to be explicitly said though because "public API functions" could reasonably be interpreted as anything in the docs (which is not what is intended I think).

Reply:
~We defined public API in one of the other documents I think.~  Correction: we don't define it.

It might be worthwhile thinking about having a policy definitions document which other policies can reference.  We definitely do not want to end up defining a term in more than one place.

Reply:
> We definitely do not want to end up defining a term in more than one place.

True. Especially if they're not the same definition! :-)

Reply:
Sooo.... what to do with this policy - should we for now add the definition of public API here and later we do a separate policy document with definitions?

Reply:
I'd be happy with adding it here for now and fixing it later. The stable release updates policy also adds a lot of definitions - so getting a standard "definitions" policy will at least need to make updates to that.

Reply:
At least one of the definitions is what is a minor release. We might want to have a policy for patch level releases, assuming it's more restrictive than that of a minor release.

Reply:
Actually there is no difference in regards to this proposal on what can be done within a minor release and in between minor releases. Any non-additions can be done only for a major release.